tests/test_views.py

```diff
@@ -740,3 +743,151 @@ def test_start_input_parameters(app, session, default_user,
             workflow = Workflow.query.filter(
                 Workflow.id_ == workflow_created_uuid).first()
             assert workflow.input_parameters == parameters['input_parameters']
+
+
+@pytest.mark.parametrize("state", [WorkflowStatus.created,
+                                   WorkflowStatus.failed,
+                                   WorkflowStatus.finished,
+                                   pytest.param(WorkflowStatus.deleted,
+                                                marks=pytest.mark.xfail),
+                                   pytest.param(WorkflowStatus.running,
+                                                marks=pytest.mark.xfail)])
+@pytest.mark.parametrize("hard_delete", [True, False])
+def test_delete_workflow(app,
+                         session,
+                         default_user,
+                         sample_yadage_workflow_in_db,
+                         state,
+                         hard_delete):
+    """Test deletion of a workflow in all possible states."""
+    sample_yadage_workflow_in_db.status = state
+    session.add(sample_yadage_workflow_in_db)
+    session.commit()
+
+    _delete_workflow(sample_yadage_workflow_in_db, hard_delete=hard_delete)
```

Comment:
Should we use the REST API to do this instead of calling directly `_delete_workflow`? (in this and all the following tests). So we test our view logic and make sure that status responses and messages are correct? It is a place that we could "break", specially since we are aggregating all status operations in a single endpoint.

Reply:
@diegodelemos please take a look